gflownet/envs/crystals/composition.py

```diff
@@ -617,16 +617,28 @@ def _can_produce_neutral_charge(self, state: Optional[List[int]] = None) -> bool
             for i, num in enumerate(state)
             if num > 0
         ]
-        sum_diff_elem = []
 
-        for n, c in nums_charges:
-            charge_sums = []
-            for c_i in itertools.product(c, repeat=n):
-                charge_sums.append(sum(c_i))
-            sum_diff_elem.append(np.unique(charge_sums))
-
-        poss_charge_sum = [
-            sum(combo) == 0 for combo in itertools.product(*sum_diff_elem)
-        ]
```

Comment:
So overall the PR looks good to me, thank you for this! One thing I was wondering about: I noticed now that this is terribly inefficient in the worst case, and should be something like:
```
for combo in itertools.product(*sum_diff_elem):
  if sum(combo) == 0:
    return True
return False
```
Did you/could you benchmark against something like this as well? I don't think early exit would be possible with your proposed implementation.

Reply:
I actually just checked it - still terribly slow. Approving!